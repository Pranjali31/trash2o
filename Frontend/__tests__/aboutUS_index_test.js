import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import AboutUs from '../src/screens/aboutUs/index';

test("AboutUs Component Loading", () => {
    render(<AboutUs />)

})
test("index component snapshot", () => {
    let component = render(<AboutUs />).toJSON();
    expect(component).toMatchSnapshot();
    console.log(JSON.stringify(component))
})


describe("index Testing component", () => {
    var component;
    beforeEach(() => {
        component = render(<AboutUs />).toJSON();
    })

    test("Testing about_us_1", () => {
        let styles = component.children[0].children[0].children[0].children[0];
        expect(styles).toStrictEqual('Tras.h2O is a one stop solution for getting to know the refill stations that are avaible nearby as well as recording the data of the usage and frequency of the refilling ,For each activity perfomred few points are added into the users account which could be used to redeem for a coupon from the ones that are availble.');
    })

    test("Testing about_us_2", () => {
        let styles = component.children[0].children[0].children[1].children[0];
        expect(styles).toStrictEqual("Tras.h2O's objective is to provide a platform and service that could contribute to the reusability of containers as well as motivate users to go green as well as encourage others to join in.");
    })

    test("Testing about_us_3", () => {
        let styles = component.children[0].children[0].children[2].children[0];
        expect(styles).toStrictEqual('Get to Know our team: ');
    })

    test("Testing about_us_4", () => {
        let styles = component.children[0].children[0].children[3].children[0];
        expect(styles).toStrictEqual('Ayush Kaushik Mistry ');
    })

    test("Testing about_us_5", () => {
        let styles = component.children[0].children[0].children[4].children[0];
        expect(styles).toStrictEqual('Harsh Bharath Nimmakuri ');
    })

    test("Testing about_us_6", () => {
        let styles = component.children[0].children[0].children[5].children[0];
        expect(styles).toStrictEqual('Karan Amitkumar Zaveri');
    })

    test("Testing about_us_7", () => {
        let styles = component.children[0].children[0].children[6].children[0];
        expect(styles).toStrictEqual('Marc Smith');
    })

    test("Testing about_us_7", () => {
        let styles = component.children[0].children[0].children[7].children[0];
        expect(styles).toStrictEqual('Parth Ashokbhai Patel ');
    })

    test("Testing about_us_7", () => {
        let styles = component.children[0].children[0].children[8].children[0];
        expect(styles).toStrictEqual('Pranjali Mukeshkumar Prajapati');
    })

    test("about_us Testing Styles1", () => {
        const index = component
        // console.log(JSON.stringify(homescreen));
        let Styles = index.children[0].children[0].children[0].props.style;
        expect(Styles).toStrictEqual({
            paddingHorizontal: 10,
    fontStyle: 'italic',
    fontWeight: 'bold',
    fontSize: 20,
    lineHeight: 30,
        });
    });
    test("about us Testing Styles2", () => {
        const index = component
        // console.log(JSON.stringify(homescreen));
        let Styles = index.children[0].children[0].children[8].props.style;
        expect(Styles).toStrictEqual({
            paddingTop: 20,
            paddingHorizontal: 40,
            fontStyle: 'italic',
            fontWeight: 'bold',
            fontSize: 20,
            lineHeight: 30,
        
        });
    });

    test("about us Testing Styles3", () => {
        const index = component
        // console.log(JSON.stringify(homescreen));
        let Styles = index.props.style;
        expect(Styles).toStrictEqual({
            paddingVertical: 20,
            backgroundColor: '#050f24',
            flex: 1,
            paddingHorizontal: 10,
        });
    });

    test("about us Testing Styles4", () => {
        const index = component
        // console.log(JSON.stringify(homescreen));
        let Styles = index.children[0].children[0].props.style;
        expect(Styles).toStrictEqual({
            flex: 1,
    paddingBottom: 100,
        });
    });

})

