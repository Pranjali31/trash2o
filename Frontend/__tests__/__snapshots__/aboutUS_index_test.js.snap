// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index component snapshot 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#050f24",
      "flex": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingBottom": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 10,
          }
        }
      >
        Tras.h2O is a one stop solution for getting to know the refill stations that are avaible nearby as well as recording the data of the usage and frequency of the refilling ,For each activity perfomred few points are added into the users account which could be used to redeem for a coupon from the ones that are availble.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 10,
            "paddingTop": 20,
          }
        }
      >
        Tras.h2O's objective is to provide a platform and service that could contribute to the reusability of containers as well as motivate users to go green as well as encourage others to join in.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 10,
            "paddingTop": 20,
          }
        }
      >
        Get to Know our team: 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 40,
            "paddingTop": 20,
          }
        }
      >
        Ayush Kaushik Mistry 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 40,
            "paddingTop": 20,
          }
        }
      >
        Harsh Bharath Nimmakuri 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 40,
            "paddingTop": 20,
          }
        }
      >
        Karan Amitkumar Zaveri
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 40,
            "paddingTop": 20,
          }
        }
      >
        Marc Smith
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 40,
            "paddingTop": 20,
          }
        }
      >
        Parth Ashokbhai Patel 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "fontWeight": "bold",
            "lineHeight": 30,
            "paddingHorizontal": 40,
            "paddingTop": 20,
          }
        }
      >
        Pranjali Mukeshkumar Prajapati
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
